tests/tests_switch_provider.yml

```
@@ -6,5 +6,7 @@
 - import_playbook: playbooks/tests_switch_provider.yml
   when:
     # The test requires or should run with NetworkManager, therefore it cannot
-    # run on RHEL/CentOS 6
-    - ansible_distribution_major_version != '6'
+    # run on RHEL/CentOS 6. In addition it cannot run on CentOS/RedHat 9.
+    - ansible_distribution in ['centOS', 'RedHat'] and
+      ansible_distribution_major_version != '6' and
+      ansible_distribution_major_version | int < 9
```

I guess I was not clear enough, my suggestion is this:

```suggestion
    # run on RHEL/CentOS 6 or CentOS/RedHat 9 and later.
    - ansible_distribution_major_version != '6' and
      ansible_distribution_major_version | int < 9
```

---

@liangwen12year properly noticed that this change will skip the test also on Fedora, which is wrong. I will revert it.